txpipe/twopoint_null_tests.py

```diff
@@ -766,3 +767,272 @@ def write_output(self, source_list, lens_list, meta, results):
+            
+            else:
+                heb = hybrideb.HybridEB(self.config['theta_min'], self.config['theta_max'], self.config['Ntheta'])
+                beb = hybrideb.BinEB(self.config['theta_min'], self.config['theta_max'], self.config['Ntheta'])
+                geb = hybrideb.GaussEB(beb, heb)
+                geb_dict = {}
+                for i in range(0,self.config['Nell']):
+                    geb_dict['%d'%(i+1)] = {}
+                    for j in range(0,6):
+                        geb_dict['%d'%(i+1)]['%d'%(j+1)]=geb(i)[j]
+            
+                with open(file_precomputed_weights, 'wb') as handle:
+                    pickle.dump(geb_dict, handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+            En0  = np.zeros(self.config["Nell"])
+            Bn0  = np.zeros(self.config["Nell"])
+            En   = np.zeros((self.config["Nell"],self.config["Nsims"]))
+            Bn   = np.zeros((self.config["Nell"],self.config["Nsims"]))
+            ell  = np.zeros(self.config["Nell"])
+
+            filename = self.get_input("twopoint_data_real_raw")
+            s    = sacc.Sacc.load_fits(filename)
+
+            Qxip = sacc.standard_types.galaxy_shear_xi_plus
+
+            source_tracers = set()
+            for b1, b2 in s.get_tracer_combinations(Qxip):
+                source_tracers.add(b1)
+                source_tracers.add(b2)
+
+            nbin_source = max(len(source_tracers), 1)
+
+            corvarr = np.zeros( (int(nbin_source*(nbin_source+1)/2*self.config["Nell"]), self.config["Nsims"])  )
+
+            ret=np.zeros((nbin_source,nbin_source,self.config["Nell"]))
+
+            c = 0
+            for zi in range(0,nbin_source):
+                for zj in range(zi,nbin_source):
+                    ii = self.config["Nell"]*c
+                    ff = self.config["Nell"]*(c+1)
+                    
+                    # Need all 4 block of the covariance
+                    filename = self.get_input("twopoint_data_real_raw")
+                    tmp      = sacc.Sacc.load_fits(filename)
+                    
+                    # From the sacc file only load relevant tracer combinations 
+                    tmp.keep_selection(tracers=('source_%d'%zj, 'source_%d'%zi))
+                    dvec = tmp.mean
+                    xip  = dvec[:int(self.config["Ntheta"])]
+                    xim  = dvec[int(self.config["Ntheta"]):]
+
+                    #random draws based on mean dn covariance
+                    x = np.random.multivariate_normal(mean = tmp.mean, cov =tmp.covariance.covmat , size = self.config["Nsims"])
+                    Rxip = x[:,:int(self.config["Ntheta"])]
+                    Rxim = x[:,int(self.config["Ntheta"]):]
+
+                    # Compute the uncertainties from random draws          
+                    for n in range(self.config["Nsims"]):
+                        for i in range(int(self.config["Nell"])):
+                            res     = geb_dict['%d'%(i+1)]   
```

Comment:
can do f"{i+1}" these days to be slightly more readable

Reply:
Fixed.